File: general_scripts/template.pl
#!/usr/bin/perl -w

# 2011 (C) Karolis Uziela, karolis.uziela@cs.helsinki.fi
# Antti Honkela group
# University of Helsinki
# Helsinki, Finland

# Script description:
# N/A


use strict;

if ($#ARGV != 1) {
	die "
Usage:

$0 [Parameters]

Parameters:
  <input-file> - input file
  <output-file> - output file
\n";
}

# input files/directories	
my $INPUT_FILE = $ARGV[0];

# output files/directories
my $OUTPUT_FILE = $ARGV[1];

# constants
# N/A

# global variables
# N/A

###################### main script ######################

print "$0 has started with parameters: @ARGV\n";


print "$0 done.\n";

###################### subroutines ######################

sub read_file {
	my $input_file = $_[0];
	open(IN_FILE, "$input_file") or die "Error occured opening input file '$input_file': $!";

	#my @file = <IN_FILE>;
	#for (my $i = 0; $i <= $#file; $i++) {
	#}

	while (my $line=<IN_FILE>) {
		chomp $line;
	}

	close(IN_FILE) or die "Error occured closing input file: $!";
}

sub write_file {
	my $output_file = $_[0];
	open(OUT_FILE,">$output_file") or die "Error occured opening output file: '$output_file': $!";
	
	close(OUT_FILE) or die "Error occured closing output file: $!";
}

File: general_scripts/template.py
#!/usr/bin/env python

# Author: Karolis Uziela

import sys

################################ Usage ################################

script_name = sys.argv[0]

usage_string = """
Usage:

	%s [Parameters]
	
Parameters:

	<input-file> - input file
	<output-file> - output file
""" % script_name

if len(sys.argv) != 3:
    sys.exit(usage_string)

################################ Functions ################################

def read_data(filename):
    
    f = open(filename)
    
    while True:
        line = f.readline()
        if len(line) == 0: 
            break
        line = line.rstrip('\n')
        #bits = line.split("\t")     
        
    f.close()
    
def write_data(filename):
	f = open(output_file,"w")
	
	#out_str = "%s %f \n" % str_var f_var
	#f.write(out_str)
	
	f.close()


################################ Variables ################################

# Input files/directories
input_file = sys.argv[1]

# Output files/directories
output_file = sys.argv[2]

# Constants
# N/A

# Global variables
# N/A

################################ Main script ################################
	
print "%s is running..." % script_name



print "%s done." % script_name



